### PR TITLE
docs: istio compatibility

### PR DESCRIPTION
**Pull request by emil2k**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

- TKC-2018

### CODE REVIEW

File: docs/docs/articles/istio.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# Using Istio
```
Comment (dejanzele) [inline docs/docs/articles/istio.md:1]: I'd change the structure a bit.

You can start with a TL;DR; on Istio but that is completely optional.

I'd first start with introducing the problem statement with Testkube & Istio and then introduce a solution, currently it is hard to immediatelly get context what is the problem statement.

Context - optional

Problem Stamenent

Solution

Examples

Testkube compatibility (mention what is officially supported and what not)

File: docs/docs/articles/istio.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    testWorkflows:
+        globalTemplate:
+            enabled: true
+            spec:
+                pod:
+                    annotations:
+                        sidecar.istio.io/inject: "false"
+```
+
+#### Hold the API server till Istio's proxy is ready
```
Comment (dejanzele) [inline docs/docs/articles/istio.md:98]: Mention also setting the `holdApplicationUntilProxyStarts` in Istio config

File: docs/docs/articles/istio.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+Installations that do not currently support native sidecars should apply the
+configurations below to utilize Testkube until they can upgrade their cluster to
+enable native sidecars:
+
+### Configurations for Istio installations without native sidecars
+
+All the values mentioned will be from the top of the specified chart.
+
+#### Disable Istio for test executions of prebuilt/container executors
```
Comment (dejanzele) [inline docs/docs/articles/istio.md:45]: I'd keep the heading name short and add the explanation below instead of just `Chart `testkube``

**Comment by dejanzele** [comment]
@jfermi can you also please take a look?

File: docs/docs/articles/istio.md (hunk 2 above)
Comment (emil2k) [reply docs/docs/articles/istio.md:98]: We are setting it at the pod level below, setting it globally is not necessary and does not seem to be recommended from the discussions I have read.

File: docs/docs/articles/istio.md (hunk 1 above)
Comment (emil2k) [reply docs/docs/articles/istio.md:1]: The problems are stated below, and the context is required to understand them. These are not Testkube-specific problems.

I think this fine, unless you have some concrete suggestions.

**Review by jfermi — COMMENTED** [review]
Editing suggestion.

File: docs/docs/articles/istio.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    ready](https://github.com/istio/istio/issues/11130), otherwise, network requests
+    may fail.
+-   Containers, [especially within batch
+    jobs](https://github.com/istio/istio/issues/6324), must signal to the proxy
+    that they have completed, otherwise, the proxy never exits and the job never
+    completes.
+
+## Compatibility with Istio
+
+Testkube has been verified to work Istio installations utilizing native
```
Comment (emil2k) [inline docs/docs/articles/istio.md:32]:
```suggestion
Testkube has been verified to work with Istio installations utilizing native
```

**Review by vsukhin — APPROVED** [review]

**Pull request merged**